tests/test_elastic.py

```diff
@@ -743,5 +743,215 @@ def test_multiple_migrations_with_delete(self):
+        elastic_search.delete_file(self.fake_file_doc_one['path'])
+        res = query('test file')['results']
+        assert_equal(len(res), 0)
+
+    def test_remove_all_file(self):
+        res_one = query('test file')['results']
+        res_two = query('spain rains')['results']
+        assert_equal(len(res_one), 1)
+        assert_equal(len(res_two), 1)
+
+        file_list = [self.fake_file_doc_one, self.fake_file_doc_two]
+        file_dict_list = []
+        for f in file_list:
+            f_dict = f.copy()
+            del f_dict['content']
+            del f_dict['parent_id']
+            f_dict.update({'addon': 'some_addon'})
+            file_dict_list.append(f_dict)
+
+        with mock.patch('website.search.file_util.collect_files', return_value=file_dict_list):
+            elastic_search.delete_all_files(self.parent_project)
+        time.sleep(1)
+
+        res_one = query('test file')['results']
+        res_two = query('spain rains')['results']
+        assert_equal(len(res_one), 0)
+        assert_equal(len(res_two), 0)
+
+
+class TestSearchUpdate(SearchTestCase):
+    def setUp(self):
+        super(TestSearchUpdate, self).setUp()
+        self.project = ProjectFactory()
+        self.project.set_privacy('public')
+
+    @mock.patch('website.project.model.Node.update_search_files')
+    def test_update_called_on_make_public(self, mock_update_search_files):
+        self.project.set_privacy('private')
+        mock_update_search_files.reset_mock()
+        assert_false(mock_update_search_files.called)
+        self.project.set_privacy('public')
+        assert_true(mock_update_search_files.called)
+
+    @mock.patch('website.project.model.Node.update_search_files')
+    def test_update_called_on_make_private(self, mock_update_search_files):
+        assert_false(mock_update_search_files.called)
+        self.project.set_privacy('private')
+        assert_true(mock_update_search_files.called)
+
+    @mock.patch('website.search.search.update_file')
+    def test_update_called_on_file_created(self, mock_update_file):
+        assert_false(mock_update_file.called)
+        self.project.update_search_file('create', 'some addon', 'some name', 'some path')
+        assert_true(mock_update_file.called)
+
+    @mock.patch('website.project.model.Node.update_search_file')
+    def test_update_called_on_file_update(self, mock_update_file):
+        assert_false(mock_update_file.called)
+        self.project.update_search_file('update', 'some addon', 'some name', 'some path')
+        assert_true(mock_update_file.called)
+
+    @mock.patch('website.project.model.Node.update_search_file')
+    def test_remove_called_on_file_delete(self, mock_remove_file):
+        assert_false(mock_remove_file.called)
+        self.project.update_search_file('delete', 'some addon', 'some name', 'some path')
+        assert_true(mock_remove_file.called)
+
+
+@unittest.skip('Travis dosn\'t seem to like loading files in tests.')
```

Comment:
No need to change this, but I would typically have used doublequotes for the string, instead of escaping the apostrophe.